app/controllers/books_controller.rb

```diff
@@ -70,6 +76,17 @@ def set_book
 
     # Never trust parameters from the scary internet, only allow the white list through.
     def book_params
-      params.require(:book).permit(:title, :author, :edition, :year, :isbn)
+      params.require(:book).permit(:title, :author, :edition, :year, :isbn, :user, :status)
+    end
+
+    def check_for_reservation_changes(old_status, new_status)
+      status_change = [old_status, new_status]
+
+      if status_change == [Book::AVAILABLE, Book::RESERVED]
```

Comment:
We could refactor this to use a state machine gem which would make it a bit easier to manage when we get more state transitions